crypto/s2n_pkey_mldsa.c

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "crypto/s2n_pkey_mldsa.h"
+
+#include "crypto/s2n_pkey_evp.h"
+#include "utils/s2n_safety.h"
+
+bool s2n_pkey_mldsa_supported()
+{
+#if S2N_LIBCRYPTO_SUPPORTS_MLDSA
+    return true;
+#else
+    return false;
+#endif
+}
+
+int s2n_pkey_mldsa_sign(const struct s2n_pkey *priv_key, s2n_signature_algorithm sig_alg,
+        struct s2n_hash_state *digest, struct s2n_blob *signature)
+{
+    POSIX_ENSURE(s2n_in_unit_test(), S2N_ERR_UNIMPLEMENTED);
```

Comment:
nit: is it possible for these errors to actually be encountered if an ML-DSA cert is received? If so should these maybe be protocol errors like `S2N_ERR_CERT_TYPE_UNSUPPORTED`? Or should `s2n_pkey_mldsa_init` just raise the error maybe?

Reply:
It is possible: if the application tries to load an ML-DSA cert and the libcrypto supports ML-DSA, then the failure happens during the s2n_pkey_match signature check. I wasn't considering the case where the peer sends an ML-DSA cert, because that isn't currently possible until we actually support the right siganture schemes. So "protocol error" isn't right, but I guess "internal error" isn't ideal either.

I could switch it to S2N_ERR_INVALID_PEM, which is a usage error. I don't want to make a new error bc this is temporary. I could also just remove the error-- it's not really the end of a the world if a curious customer poking around in the code decides to load an ML-DSA cert. They can't do anything with it yet.

Reply:
I'm going to go with just removing the error.